### PR TITLE
Add gas per second metrics support

### LINKED ISSUE #5052: HIP-584: Implement gas limit strategy for single `eth_call` request
### Problem

As additional security layer for the `Mirror Archive Nodes` we can limit the execution of a single `eth_call` request by a given gas limit value. This would help us protect the system from processing infinity loops by the `EVM library` internally halting the execution when the given gas limit is reached. Even though only read-only operations would be executed we can still count on the gas usage that is accumulated during the execution and have an upper limit that the EVM would use.

### Solution

Define a proper value for a default `gasLimit` value for a single `eth_call` request and pass it inside the `EVM` during building the `EVM` instance. 

Define a proper error handling strategy when the gas limit is reached and the request fails.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:

Add custom metrics for gas per second inside web3 module. The metric is exposed to `MetricRegistry` under this name - `hedera_mirror_web3_call_gas`

**Related issue(s)**:

Fixes #5052 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/config/GasPerSecondMetrics.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class GasPerSecondMetrics implements MeterBinder {
+
+    private final GasPerSecond gasPerSecond;
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        Gauge.builder("hedera.mirror.web3.call.gas", gasPerSecond, GasPerSecond::getAccumulatedGas) //
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/config/GasPerSecondMetrics.java:35]: Are there any tags we should add to differentiate different types of EVM calls? At least we should have estimate vs not estimate calls. Anything else?

Note that if we have dynamic tags we have to use a concurrent map to lazy create them. See `Downloader.nodeSignatureStatusMetricMap` for an example.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/traceability/GasMetricsHederaTracer.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public void tracePostExecution(
+            final MessageFrame currentFrame, final OperationResult operationResult) {
+        long postExecutionRemainingGas = currentFrame.getRemainingGas();
+
+        long gasUsedForOperation = preExecutionRemainingGas - postExecutionRemainingGas;
+
+        gasPerSecond.incrementGas(gasUsedForOperation);
+    }
+}
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/traceability/GasMetricsHederaTracer.java:52]: But the above is probably too slow to do per operation. Instead, we should move the above counter to `ContractCallService` and `counter.increment(hederaEvmTransactionProcessingResult.getGasUsed())`.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/config/GasPerSecondMetrics.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/config/GasPerSecondMetrics.java]: Changed to Counter.

Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/config/GasPerSecondMetrics.java]: That's a great idea! Due to the limited time-frame, I suggest enhancing the custom metric with these dynamic tags next sprint.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
LGTM

**Review by david-bakin-sl — APPROVED** [review]
LGTM! (have to look at this micrometer library later, it's not used in hedera-services, looks easy though!)

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**